### PR TITLE
Support K2 Analysis in DGPv2

### LINKED ISSUE #3699: [K2] Too verbose messaging about Dokka K2 in preview
```
> Configure project :
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
Dokka's K2 Analysis is being used. It is still under active development and is thus experimental. It can be the cause of failed builds or incorrectly generated documentation. If you encounter an issue, please consider reporting it: https://github.com/Kotlin/dokka/issues
```

May be we can reduce it to one message? Or at least move from the Gradle configuration phase to the dokka task execution?

### PR DESCRIPTION
**Pull request by adam-enko**
- Support toggling between K1 and K2 analysis when DGP is in V2 mode.
- Update DGPv1 to use PluginFeaturesService to query the `tryK2` flag.
- Add `org.jetbrains.dokka.experimental.tryK2.noWarn=true` to permit suppressing the K2 Analysis warning.

Depends on #3736

Fixes #3699
Fixes [KT-70659](https://youtrack.jetbrains.com/issue/KT-70659/)

### CODE REVIEW

File: dokka-runners/dokka-gradle-plugin/src/testFixtures/kotlin/projects/MultiModuleProject.kt [hunk 1]
```diff
@@ -47,6 +47,9 @@ fun TestScope.initMultiModuleProject(
             buildGradleKts = """
                 |plugins {
                 |  kotlin("jvm") version embeddedKotlinVersion
+                |  // important: Register different plugins here to make the buildscript classpath different,
```
Comment (whyoleg) [inline dokka-runners/dokka-gradle-plugin/src/testFixtures/kotlin/projects/MultiModuleProject.kt:50]: unrelated change?

**Review by whyoleg — APPROVED** [review]
LGTM

File: dokka-runners/dokka-gradle-plugin/src/testFixtures/kotlin/projects/MultiModuleProject.kt [hunk 2]
```diff
@@ -47,6 +47,9 @@ fun TestScope.initMultiModuleProject(
...
+                |  // because we _want_ the test to trigger a Gradle classloader bug.
+                |  kotlin("plugin.serialization") version embeddedKotlinVersion
```
Comment (adam-enko) [inline dokka-runners/dokka-gradle-plugin/src/testFixtures/kotlin/projects/MultiModuleProject.kt:52]:
```suggestion
                |  kotlin("jvm") version embeddedKotlinVersion
```

File: dokka-runners/dokka-gradle-plugin/src/testFixtures/kotlin/projects/MultiModuleProject.kt (hunk 1 above)
Comment (adam-enko) [reply dokka-runners/dokka-gradle-plugin/src/testFixtures/kotlin/projects/MultiModuleProject.kt]: yup, thanks for spotting it

File: dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt [hunk 3]
```diff
@@ -112,6 +118,39 @@ internal abstract class PluginFeaturesService : BuildService<PluginFeaturesServi
         }
     }
 
+    internal val enableK2Analysis: Boolean by lazy(SYNCHRONIZED) {
```
Comment (Tapchicoma) [inline dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt:121]: Nit: default for `lazy` is `SYNCHRONIZED`: https://kotlinlang.org/docs/delegated-properties.html#lazy-properties

File: dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt [hunk 4]
```diff
@@ -112,6 +118,39 @@ internal abstract class PluginFeaturesService : BuildService<PluginFeaturesServi
...
+        enableK2Analysis
+    }
+
+    private fun logK2AnalysisMessage() {
+        if (primaryService && !parameters.k2AnalysisNoWarn.getOrElse(false)) {
+            logger.warn(
+                """
+                |Dokka K2 Analysis is enabled
+                |
+                |  This feature is Experimental and is still under active development.
```
Comment (Tapchicoma) [inline dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt:139]: Curious - why 2 space indent for the major part of the message? :thinking:

**Review by Tapchicoma — APPROVED** [review]

File: dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt (hunk 3 above)
Comment (adam-enko) [reply dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt:121]: TIL. Do you think it's better to keep `SYNCHRONIZED` it to be explicit, or remove it?

File: dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt (hunk 4 above)
Comment (adam-enko) [reply dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt:139]: I have no strong reason. My intention was to differentiate between the title and body of the message, to make the message easier to scan quickly.
